spec/system/admin/raw_data_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Raw data' do
+  let(:user) { create(:user, admin: true) }
+  let(:message) { create(:message) }
+
+  it 'admin debugs raw message data' do
+    visit admin_message_path(id: message.id, as: user)
+    click_on 'text.json'
+
+    switch_to_window(windows.last)
+
+    expect(page.body).to have_text('{"text":"Hello"}')
+    # expect(page.response_headers['Content-Type']).to eq('application/json')
+    # expect(page.response_headers['Content-Disposition']).to start_with('inline;')
```

Comment:
These expectations are failing now... Not exactly sure why, but some differences I noted is that with a real browser, we must call this `switch_to_window(windows.last)` and then there are no `response_headers`, so it throws an error `NoMethod [] for NilClass`... If you refresh the page, there are `response_headers` and I could see the `Content-Disposition` did start_with `inline`, but could not see the `Content-Type`

Reply:
I've removed these expectations as they don't fit with the reality of using a real browser. See fdb29750
We could remove the `target="_blank"` from the link in [app/views/fields/raw_data_field/_show.html.rb]() and we would get the `page.response_headers` to be able to make assertions, but I'm not sure it's that important to the test.

There are other potential benefits to not opening a new window. It is an `a11y` issue, for example. Also, some believe it to be bad UX, but this is debatable. For now, I have decided to leave it as is because some admin might already be familiar with how it works now and not invite change.